client/components/button/test/index.js

```diff
@@ -0,0 +1,1064 @@
+					expect( button ).to.contain( 'Text' );
+					expect( button.is( '.button' ) ).to.equal( true );
+				} );
+
+				it( 'renders with text and icon', function() {
+					const button = shallow( <Button>{ text }<Gridicon size={ 18 } icon="arrow-left" /></Button> );
+					expect( button.find( 'button' ) ).to.contain( 'Text' );
+					expect( button.find( '.button' ) ).to.contain( icon );
+					expect( button.find( 'button' ) ).to.have.className( 'button' );
+					expect( button.find( Gridicon ) ).to.have.prop( 'icon' ).equal( 'arrow-left' );
+				} );
+
+				it( 'renders with icon', function() {
+					const button = shallow( <Button><Gridicon size={ 18 } icon="arrow-left" /></Button> );
+					expect( button.find( 'button' ) ).to.have.className( 'button' );
+					expect( button.find( '.button' ) ).to.contain( icon );
+					expect( button.find( Gridicon ) ).to.have.prop( 'icon' ).equal( 'arrow-left' );
+				} );
+
+				it( 'renders disabled', function() {
+					const button = shallow( <Button disabled>{ disabled }</Button> );
+					expect( button.find( '.button' ) ).to.be.disabled();
+				} );
+
+				describe( 'is-scary', function() {
+					it( 'renders with text', function() {
+						const button = shallow( <Button scary={ true }>{ text }</Button> );
+						expect( button ).to.contain( 'Text' );
+						expect( button.is( '.button.is-scary' ) ).to.equal( true );
+					} );
+
+					it( 'renders with text and icon', function() {
+						const button = shallow( <Button scary={ true }>{ text } { icon }</Button> );
+						expect( button.find( 'button' ) ).to.have.text( 'Text <Gridicon />' );
+						expect( button.find( '.button' ) ).to.contain( icon );
+						expect( button.find( 'button' ) ).to.have.className( 'button' );
+						expect( button.find( 'button' ) ).to.have.className( 'is-scary' );
+						expect( button.find( Gridicon ) ).to.have.prop( 'icon' ).equal( 'arrow-left' );
+					} );
+
+					it( 'renders with icon', function() {
+						const button = shallow(
+							<Button scary={ true }>
+								<Gridicon size={ 18 } icon="arrow-left" />
+							</Button>
+						);
+						expect( button.find( 'button' ) ).to.have.className( 'button' );
+						expect( button.find( 'button' ) ).to.have.className( 'is-scary' );
+						expect( button.find( '.button' ) ).to.contain( icon );
+						expect( button.find( Gridicon ) ).to.have.prop( 'icon' ).equal( 'arrow-left' );
+					} );
+
+					it( 'renders disabled', function() {
+						const button = shallow( <Button scary={ true } disabled>{ disabled }</Button> );
+						expect( button.find( '.button.is-scary' ) ).to.be.disabled();
+					} );
+				} );
+			} );
+
+			describe( 'is-primary', function() {
+				it( 'renders with text', function() {
+					const button = shallow( <Button primary={ true }>{ text }</Button> );
+					expect( button ).to.contain( 'Text' );
+					expect( button.is( '.button.is-primary' ) ).to.equal( true );
+				} );
+
+				it( 'renders with text and icon', function() {
```

Comment:
As above, I don't think we need more than one test for checking basic render behavior.